lib/ClangImporter/ImportName.cpp

```diff
@@ -2237,9 +2237,44 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
                 return;
               }
             }
-          } else if (auto namedArg = dyn_cast_or_null<clang::NamedDecl>(ty->getAsTagDecl())) {
-            importNameImpl(namedArg, version, clang::DeclarationName()).getDeclName().print(buffer);
-            return;
+          } else {
```

Comment:
This whole thing is asking to be factored out/refactored. But non blocking for this patch.

Reply:
yep, agree